lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -696,7 +696,7 @@ def compare(self, os):
             differences.add('reserve_memory', parameter=self.reserve_memory, active=os.reserve_memory)
         if self.container_labels != os.container_labels:
             differences.add('container_labels', parameter=self.container_labels, active=os.container_labels)
-        if self.publish != os.publish:
+        if self.publish and self.publish != os.publish:
```

Comment:
Why do you handle `publish` this way? If a user explicitly specifies `publish: []`, why should it behave differently than, for example, if the user explicitly specifies `networks: []`?

Reply:
This was a change reverting problematic parts of #45674 (I've open #49285 to track the problem ) I should not have pushed this here.

I'm gonna pushing the correct changes.

Reply:
Ok :)